### PR TITLE
Model optimizer 2022.1 adjustments

### PR DESCRIPTION
**Pull request by mkuczyns11**
In previous versions of model optimizer east_fp32 model's input and outputs had NCHW layout after optimizing. Since 2022.1 model optimizer does not change the layout. Therefore east_ocr custom node was adjusted to expect all inputs in NHWC layout. If you intend to use east_fp32 model optimized with pre 2022.1 model optimizer it is necessary to force NHWC layout on both inputs and outputs of east_fp32 model.

### CODE REVIEW

File: src/custom_nodes/east_ocr/east_ocr.cpp [hunk 1]
```diff
@@ -247,11 +247,7 @@ int execute(const struct CustomNodeTensor* inputs, int inputsCount, struct Custo
     // Extract the scores (probabilities), followed by the geometrical data used to derive potential bounding box coordinates that surround text
     for (int y = 0; y < numRows; y++) {
         float* scoresData = (float*)scoresTensor->data + (y * numCols);
-        float* xData0 = (float*)geometryTensor->data + ((0 * numRows * numCols) + (y * numCols));
-        float* xData1 = (float*)geometryTensor->data + ((1 * numRows * numCols) + (y * numCols));
-        float* xData2 = (float*)geometryTensor->data + ((2 * numRows * numCols) + (y * numCols));
```
Comment (dkalinowski) [inline src/custom_nodes/east_ocr/east_ocr.cpp:252]: Why do you remove it?

File: demos/optical_character_recognition/python/README.md [hunk 2]
```diff
@@ -68,13 +68,13 @@ export_model('./east_icdar2015_resnet_v1_50_rbox/model.ckpt-49491','./model.pb')
...
 ```bash
-docker run -u $(id -u):$(id -g) -v ${PWD}/:/EAST:rw -w /EAST openvino/ubuntu18_dev:2021.3 python3 freeze_east_model.py
+docker run -u $(id -u):$(id -g) -v ${PWD}/:/EAST:rw -w /EAST tensorflow/tensorflow:1.15.5 python3 freeze_east_model.py
 ```
 
 Convert the TensorFlow frozen model to Intermediate Representation format using the model_optimizer tool:
 ```bash
-docker run -u $(id -u):$(id -g) -v ${PWD}/:/EAST:rw openvino/ubuntu18_dev:2021.3 deployment_tools/model_optimizer/mo.py \
---framework=tf --input_shape=[1,1024,1920,3] --input=input_images --output=feature_fusion/Conv_7/Sigmoid,feature_fusion/concat_3  \
+docker run -u $(id -u):$(id -g) -v ${PWD}/:/EAST:rw nncv-harbor.inn.intel.com/openvino/ubuntu20_dev:2022.1.0.579 python3 /usr/local/lib/python3.8/dist-packages/openvino/tools/mo/mo.py \
```
[the remark this answers is not in the document]
Comment (dtrawins) [reply demos/optical_character_recognition/python/README.md:76]: the image should be openvino/ubuntu20_dev:2022.1 (to be retested after publication)

File: src/custom_nodes/east_ocr/east_ocr.cpp [hunk 3]
```diff
@@ -407,19 +403,19 @@ int getInputsInfo(struct CustomNodeTensorInfo** info, int* infoCount, const stru
     (*info)[1].dims = (uint64_t*)malloc((*info)->dimsCount * sizeof(uint64_t));
     NODE_ASSERT(((*info)[1].dims) != nullptr, "malloc has failed");
     (*info)[1].dims[0] = 1;
-    (*info)[1].dims[1] = 1;
```
Comment (dkalinowski) [inline src/custom_nodes/east_ocr/east_ocr.cpp:410]: This custom node will not work with setups prior v2022.1. Should we notify customers?

File: src/custom_nodes/east_ocr/east_ocr.cpp [hunk 4]
```diff
@@ -225,17 +225,17 @@ int execute(const struct CustomNodeTensor* inputs, int inputsCount, struct Custo
     NODE_ASSERT(image.cols == imageWidth, "Mat generation failed");
     NODE_ASSERT(image.rows == imageHeight, "Mat generation failed");
 
-    uint64_t _numRows = scoresTensor->dims[2];
-    uint64_t _numCols = scoresTensor->dims[3];
+    uint64_t _numRows = scoresTensor->dims[1];
```
Comment (dtrawins) [inline src/custom_nodes/east_ocr/east_ocr.cpp:228]: why custom node is changing?

**Comment by mkuczyns11** [comment]
In previous versions of model optimizer east_fp32 model's input and outputs had NCHW layout after optimizing. Since 2022.1 model optimizer does not change the layout. Therefore east_ocr custom node was adjusted to expect all inputs in NHWC layout. If you intend to use east_fp32 model optimized with pre 2022.1 model optimizer it is necessary to force NHWC layout on both inputs and outputs of east_fp32 model.

File: demos/optical_character_recognition/python/README.md [hunk 5]
```diff
@@ -84,9 +84,9 @@ model.mapping
 model.xml
 ```
 Converted east-reasnet50 model will have the following interface:
-- Input name: `input_images` ; shape: `[1 3 1024 1920]` ; precision: `FP32`, layout: `N...`
-- Output name: `feature_fusion/Conv_7/Sigmoid` ; shape: `[1 1 256 480]` ; precision: `FP32`
-- Output name: `feature_fusion/concat_3` ; shape: `[1 5 256 480]` ; precision: `FP32`
+- Input name: `input_images` ; shape: `[1 1024 1920 3]` ; precision: `FP32`, layout: `N...`
+- Output name: `feature_fusion/Conv_7/Sigmoid` ; shape: `[1 256 480 1]` ; precision: `FP32`
+- Output name: `feature_fusion/concat_3` ; shape: `[1 256 480 5]` ; precision: `FP32`
```
Comment (dkalinowski) [inline demos/optical_character_recognition/python/README.md:89]: Missing layout info, please rerun to get full logs

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**